### PR TITLE
V2/check renders

### PR DESCRIPTION
**Pull request by dospore**
Couple of things were causing some slow rendering of the screen making it appear a bit janky
- move store udpate hooks into their own component (instead of _app) to avoid multiple rerenders of top level app. It doesnt actually matter where these hooks go.
- use zustand shallow where necessary. By default zustand uses === for equality so wont work completely as epected for selecting multiple state elements
- wrap functions in useCallback to memoize inner functions such that they dont cause component renders (this one doesnt effect it to much and only did for pools page)

Screen Shot 2022-04-07 at 3 39 21 pm
Screen Shot 2022-04-07 at 3 39 14 pm

### CODE REVIEW

File: archetypes/Pools/PoolDetailsModal/index.tsx [hunk 1]
```diff
@@ -21,37 +21,40 @@ type PoolProps = {
...
+            },
+            {
+                name: 'Collateral Asset',
+                value: collateralAsset,
+                href: constructExplorerLink(BlockExplorerAddressType.token, collateralAssetAddress, network),
+            },
+            {
+                name: 'Deployer',
+                value: formatAddress(committer),
+                href: constructExplorerLink(BlockExplorerAddressType.address, committer, network),
```
Comment (ScaredIbis) [inline archetypes/Pools/PoolDetailsModal/index.tsx:48]: not part of this change set but not sure what this is supposed to be (deployer or committer)

File: archetypes/Pools/index.tsx [hunk 2]
```diff
@@ -43,54 +44,69 @@ export const Browse: React.FC = () => {
...
-        .filter(searchFilter);
-    const sortedFilteredTokens = filteredTokens.sort(sorter);
+    const filteredTokens = useMemo(
+        () =>
+            tokens
+                .filter((pool) => marketFilter(pool.name, state.marketFilter))
+                .filter(leverageFilter)
+                .filter(searchFilter)
+                .sort(sorter),
+        [state.marketFilter, state.leverageFilter, state.search, state.sortBy, tokens],
```
Comment (ScaredIbis) [inline archetypes/Pools/index.tsx:93]: should this have `pool.name`, `state.marketFilter`, `leverageFilter`, `searchFilter` and `sorter` as deps instead? looks like it will work but imagining dependencies as a graph this would be a child instead of a siblings to the filters/sorter

File: hooks/useUpdatePoolInstances/index.ts [hunk 3]
```diff
@@ -21,13 +21,16 @@ import { fetchPendingCommits, V2_SUPPORTED_NETWORKS } from '~/utils/tracerAPI';
...
-    const poolAddresses = useStore(selectAllPoolLists);
-    const pools = useStore(selectPoolInstances, shallow);
+    const { setMultiplePools, resetPools, setPoolsInitialized, setTokenBalances } = useStore(
+        selectPoolInstanceActions,
+        shallow,
+    );
+    const { updateTokenApprovals, updateTokenBalances } = useStore(selectPoolInstanceUpdateActions, shallow);
+    const { addCommit } = useStore(selectUserCommitActions, shallow);
+    const { provider, account } = useStore(selectWeb3Info, shallow);
+    const poolAddresses = useStore(selectAllPoolLists, (oldState, newState) => oldState.length === newState.length);
```
Comment (ScaredIbis) [inline hooks/useUpdatePoolInstances/index.ts:31]: unlikely to happen but if switching between networks and both happen to have the same number of pools does will this get handled somewhere?

File: hooks/useUpdatePoolInstances/index.ts [hunk 4]
```diff
@@ -125,8 +127,8 @@ export const useUpdatePoolInstances = (): void => {
     }, [provider, poolsInitialized]);
 
     // update token balances and approvals when address changes
-    useEffect(() => {
-        if (provider && account && poolsInitialized) {
+    useMemo(() => {
```
Comment (ScaredIbis) [inline hooks/useUpdatePoolInstances/index.ts:130]: why switch to useMemo here? my understanding is useMemo is used like useCallback but for other types of variables

File: store/PoolInstancesSlice/index.tsx [hunk 5]
```diff
@@ -31,6 +31,31 @@ export const createPoolsInstancesSlice: StateSlice<IPoolsInstancesSlice> = (set,
             };
         });
     },
+    setMultiplePools: (pools) => {
+        const now = Date.now() / 1000;
```
Comment (ScaredIbis) [inline store/PoolInstancesSlice/index.tsx:35]: Math.floor?

**Review by ScaredIbis — COMMENTED** [review]
looks mostly good, just a few q's

File: archetypes/Pools/PoolDetailsModal/index.tsx (hunk 1 above)
Comment (dospore) [reply archetypes/Pools/PoolDetailsModal/index.tsx:48]: probably deployer, but is showing committer

File: hooks/useUpdatePoolInstances/index.ts (hunk 3 above)
Comment (dospore) [reply hooks/useUpdatePoolInstances/index.ts:31]: mmmmm, ill change it back, this check ended up not doing anything anyway

File: store/PoolInstancesSlice/index.tsx (hunk 5 above)
Comment (dospore) [reply store/PoolInstancesSlice/index.tsx:35]: thankyou

File: hooks/useUpdatePoolInstances/index.ts (hunk 4 above)
Comment (dospore) [reply hooks/useUpdatePoolInstances/index.ts:130]: should be useEffect

File: archetypes/Pools/index.tsx (hunk 2 above)
Comment (dospore) [reply archetypes/Pools/index.tsx:93]: pool is passed in as an arg, so it goes filter the pools, none of the filters have changed return the same thing

Comment (ScaredIbis) [reply archetypes/Pools/index.tsx:93]: is it guaranteed that `leveragedFilter`, `searchFilter` and `sorter` will have finished changing (as a result of `state.leverageFilter`, `state.search`, and `state.sortBy` changing) by the time this changes?

Comment (dospore) [reply archetypes/Pools/index.tsx:93]: yeah since they are callbacks defined above this one, theyll get updated first if any of the state changes

**Review by ScaredIbis — APPROVED** [review]

**Review by ScaredIbis — DISMISSED** [review]
